pkg/controller/oneagent/istio.go

```diff
@@ -17,105 +17,152 @@ import (
 	}
 
 	// Fetch endpoints via Dynatrace client
 	comHosts, err := dtc.GetCommunicationHosts()
 	if err != nil {
-		return err
+		logger.Error(err, "istio: failed to get communication endpoints")
+		return false, false
+	}
+
+	if upd, err := r.reconcileIstioConfigurations(logger, instance, comHosts, "communication-endpoint"); err != nil {
+		logger.Error(err, "istio: error reconciling config for communication endpoints")
+		return false, false
+	} else if upd {
+		return true, true
 	}
 
-	err = r.reconcileIstioCreateConfigurations(instance, comHosts, logger)
+	return false, true
+}
+
+func (r *ReconcileOneAgent) reconcileIstioConfigurations(logger logr.Logger, instance *dynatracev1alpha1.OneAgent,
+	comHosts []dtclient.CommunicationHost, role string) (bool, error) {
+	add, err := r.reconcileIstioCreateConfigurations(instance, comHosts, role, logger)
 	if err != nil {
-		logger.Error(err, "error reconciling Istio config")
-		return err
+		logger.Error(err, "istio: failed to create objects")
+		return false, err
 	}
 
-	r.reconcileIstioRemoveConfigurations(instance, comHosts, logger)
+	rem, err := r.reconcileIstioRemoveConfigurations(instance, comHosts, role, logger)
+	if err != nil {
+		logger.Error(err, "istio: failed to delete objects")
+		return false, err
+	}
 
-	return nil
+	return add || rem, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioRemoveConfigurations(instance *dynatracev1alpha1.OneAgent,
-	comHosts []dtclient.CommunicationHost, logger logr.Logger) {
+	comHosts []dtclient.CommunicationHost, role string, logger logr.Logger) (bool, error) {
 
 	listOps := &client.ListOptions{
 		Namespace:     instance.Namespace,
-		LabelSelector: labels.SelectorFromSet(buildLabels(instance.Name)),
+		LabelSelector: labels.SelectorFromSet(buildIstioLabels(instance.Name, role)),
 	}
 
 	seen := map[string]bool{}
 	for _, ch := range comHosts {
 		seen[istio.BuildNameForEndpoint(instance.Name, ch.Host, ch.Port)] = true
 	}
 
-	r.reconcileIstioRemoveConfiguration(instance, istio.VirtualServiceGVK, listOps, seen, logger)
-	r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	vsUpd, err := r.reconcileIstioRemoveConfiguration(instance, istio.VirtualServiceGVK, listOps, seen, logger)
+	if err != nil {
+		return false, err
+	}
+
+	seUpd, err := r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	if err != nil {
+		return false, err
+	}
+
+	return vsUpd || seUpd, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(instance *dynatracev1alpha1.OneAgent, gvk schema.GroupVersionKind,
-	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) {
+	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) (bool, error) {
 
 	var list unstructured.UnstructuredList
 	list.SetGroupVersionKind(gvk)
 
 	if err := r.client.List(context.TODO(), listOps, &list); err != nil {
-		return
+		return false, err
 	}
 
+	upd := false
```

Comment:
This var should be named `del` or similar, to not confuse with the *update* term.